node/src/BaseClient.ts

```diff
@@ -1186,18 +1208,23 @@ export class BaseClient {
      * See https://redis.io/commands/smembers/ for details.
      *
      * @param key - The key to return its members.
-     * @returns All members of the set.
-     * If `key` does not exist, it is treated as an empty set and this command returns empty list.
+     * @returns A `Set` containing all members of the set.
+     * If `key` does not exist, it is treated as an empty set and this command returns an empty `Set`.
      *
      * @example
      * ```typescript
      * // Example usage of the smembers method
      * const result = await client.smembers("my_set");
-     * console.log(result); // Output: ["member1", "member2", "member3"]
+     * console.log(result); // Output: Set {'member1', 'member2', 'member3'}
      * ```
      */
-    public smembers(key: string): Promise<string[]> {
-        return this.createWritePromise(createSMembers(key));
+    public smembers(key: string): Promise<Set<string>> {
+        return this.createWritePromise<string[]>(createSMembers(key)).then(
+            (result: string[]) => {
+                const resultSet: Set<string> = new Set<string>(result);
+                return resultSet;
+            },
+        );
```

Comment:
make a function which converts result to set and call it in `then`.
BTW, can you refer the `Set` constructor here?
Something like
```java
return this.createWritePromise<string[]>(createSMembers(key)).then(Set<string>::new);
```

Reply:
nope 😕